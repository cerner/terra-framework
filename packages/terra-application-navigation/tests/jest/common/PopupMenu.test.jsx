import React from 'react';
import PopupMenu from '../../../src/common/_PopupMenu';

describe('PopupMenu', () => {
  it('should render default element', () => {
    const shallowComponent = shallow(
      <PopupMenu />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render prop data', () => {
    const shallowComponent = shallow(
      <PopupMenu
        title="my-header"
        footerText="my-footer"
        customContent={<div>my-custom-content</div>}
        userConfig={{
          name: 'user-name',
          detail: 'user-detail',
          initials: 'user-initials',
          imageSrc: 'user-src',
        }}
        menuItems={[
          {
            text: 'test-text',
            icon: <span>my icon</span>,
            key: 'my-test-key',
            isActive: true,
            metaData: { test: 'test-data' },
            notificationCount: 2,
          },
        ]}
        onSelectMenuItem={jest.fn()}
        onSelectFooterItem={jest.fn()}
        showSelections
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });

  it('should render with height bounded', () => {
    const shallowComponent = shallow(
      <PopupMenu
        title="my-header"
        footerText="my-footer"
        isHeightBounded
      />,
    );

    expect(shallowComponent).toMatchSnapshot();
  });
});
