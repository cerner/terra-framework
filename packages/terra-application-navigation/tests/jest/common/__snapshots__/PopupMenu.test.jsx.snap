// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupMenu should render default element 1`] = `
<ContentContainer
  className="container"
  fill={false}
  footer={<ActionFooter />}
  header={
    <ActionHeader
      aria-hidden={true}
      level={1}
    />
  }
>
  <div
    className="content"
  >
    <ul
      className="utility-list"
      onKeyDown={[Function]}
      role="listbox"
      tabIndex="0"
    />
  </div>
</ContentContainer>
`;

exports[`PopupMenu should render prop data 1`] = `
<ContentContainer
  className="container"
  fill={false}
  footer={
    <ActionFooter
      end={
        <Button
          data-navigation-utility-item-logout={true}
          isBlock={false}
          isCompact={false}
          isDisabled={false}
          isIconOnly={false}
          isReversed={false}
          onClick={[MockFunction]}
          onKeyDown={[Function]}
          refCallback={[Function]}
          text="my-footer"
          type="button"
          variant="neutral"
        />
      }
    />
  }
  header={
    <ActionHeader
      aria-hidden={true}
      level={1}
      title="my-header"
    />
  }
>
  <div
    className="content"
  >
    <div
      className="custom-content-container"
    >
      <div>
        my-custom-content
      </div>
    </div>
    <PopupMenuUser
      userConfig={
        Object {
          "detail": "user-detail",
          "imageSrc": "user-src",
          "initials": "user-initials",
          "name": "user-name",
        }
      }
    />
    <ul
      aria-label="my-header"
      className="utility-list"
      onKeyDown={[Function]}
      role="listbox"
      tabIndex="0"
    >
      <PopupMenuListItem
        icon={
          <span>
            my icon
          </span>
        }
        isSelected={true}
        key="my-test-key"
        loopFocus={[Function]}
        notificationCount={2}
        onSelect={[Function]}
        parentRole="listbox"
        showSelections={true}
        text="test-text"
      />
    </ul>
  </div>
</ContentContainer>
`;

exports[`PopupMenu should render with height bounded 1`] = `
<ContentContainer
  className="container"
  fill={true}
  footer={<ActionFooter />}
  header={
    <ActionHeader
      aria-hidden={true}
      level={1}
      title="my-header"
    />
  }
>
  <div
    className="content"
  >
    <ul
      aria-label="my-header"
      className="utility-list"
      onKeyDown={[Function]}
      role="listbox"
      tabIndex="0"
    />
  </div>
</ContentContainer>
`;
